core/string/string_builder.cpp

```diff
@@ -91,9 +99,7 @@ String StringBuilder::as_string() const {
 		}
 	}
 
-	String final_string = String(buffer, string_length);
-
-	memdelete_arr(buffer);
+	String final_string = String(buffer_persist.ptr(), string_length);
 
 	return final_string;
```

Comment:
Wouldn't it be better to return the String directly?
```suggestion
	return String(buffer_persist.ptr(), string_length);
```